flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -421,6 +474,13 @@ public void disconnectJobManager(final JobID jobId, final Exception cause) {
 					slotRequest.getJobId(),
 					slotRequest.getAllocationId());
 
+				if (shouldRejectRequests()) {
```

Comment:
I think this check should rather be made in the `ResourceActionsImpl#allocateResource` because the slot request might still be fulfilled with an existing slot.